src/log/Dispatcher.php

```diff
@@ -17,21 +18,74 @@
  */
 class Dispatcher extends \yii\log\Dispatcher
 {
-    /** @since 3.6.9 */
-    public const TARGET_FILE = '__file__';
+    /** @since 4.0.0 */
+    public const TARGET_WEB = 'web';
 
-    /** @since 3.6.9 */
-    public const TARGET_STDOUT = '__stdout__';
+    /** @since 4.0.0 */
+    public const TARGET_CONSOLE = 'console';
 
-    /** @since 3.6.9 */
-    public const TARGET_STDERR = '__stderr__';
+    /** @since 4.0.0 */
+    public const TARGET_QUEUE = 'queue';
+
+    /**
+     * @var array Config to pass to each MonologTarget
+     * @since 4.0.0
+     */
+    public array $monologTargetConfig = [];
 
     /**
      * @inheritdoc
      */
     public function init(): void
     {
         parent::init();
-        $this->targets = array_merge(App::defaultLogTargets(), $this->targets);
+
+        $this->targets = array_merge($this->getTargets(), $this->targets);
+    }
+
+    /**
+     * @return MonologTarget[]
+     */
+    public function getTargets(): array
+    {
+        // Warning - Don't do anything that could cause something to get logged from here!
+        // If the dispatcher is configured with flushInterval => 1, it could cause a PHP error if any log
+        // targets haven’t been instantiated yet.
+
+        $isConsoleRequest = Craft::$app->getRequest()->getIsConsoleRequest();
+
+        // Only log console requests and web requests that aren't getAuthTimeout requests
+        if (!$isConsoleRequest && !Craft::$app->getUser()->enableSession) {
+            return [];
+        }
+
+        $targets = Collection::make([
+            static::TARGET_WEB,
+            static::TARGET_CONSOLE,
+            static::TARGET_QUEUE,
+        ])->mapWithKeys(function($name) {
+            $config = array_merge($this->monologTargetConfig, [
+                'name' => $name,
+                'enabled' => false,
+            ]);
+
+            return [$name => new MonologTarget($config)];
+        });
+
+        // Enabled via QueueLogBehavior
+        if (!YII_DEBUG) {
+            $queueTarget = $targets->get(static::TARGET_QUEUE);
+            // TODO: Ask about except/levels
+            $queueTarget->except = ['yii\*'];
+            $queueTarget->setLevels(['info', 'warning', 'error']);
```

Comment:
This was copied verbatim from [existing behavior](https://github.com/craftcms/cms/blob/4.0/src%2Fqueue%2FQueueLogBehavior.php#L103-L107), but I don't 💯 understand what we're after.

The `setLevels` may not be as relevant now that `MonologTarget::$level` can be set…